app/lib/service/entrypoint/_isolate.dart

```diff
@@ -21,329 +21,342 @@ import 'tools.dart';
 
 final _random = Random.secure();
 
-class FrontendEntryMessage {
-  final int frontendIndex;
+class EntryMessage {
   final SendPort protocolSendPort;
   final SendPort aliveSendPort;
+  final SendPort statsSendPort;
 
-  FrontendEntryMessage({
-    required this.frontendIndex,
+  EntryMessage({
     required this.protocolSendPort,
     required this.aliveSendPort,
+    required this.statsSendPort,
   });
 }
 
-class FrontendProtocolMessage {
+/// Message sent from the isolate to indicate that it is ready with the initialization.
+class ReadyMessage {
```

Comment:
You could do:
```dart
sealed class Message {}
final ReadyMessage extends Message {
  final String stuff;
  ReadyMessage(this.stuff);
}
final EntryMessage extends Message {}
final DebugMessage extends Message {}
```

then you can do:
```dart
_protocolReceivePort.listen((e) {
 if (e is! Message) {
   return;
 }
 switch(e) {
   case ReadyMessage(stuff: var s):
     print(s);
   case EntryMessage():
   case DebugMessage():
 }

});
```

No need though, just promoting new Dart 3 features :rofl:

Reply:
I think this is nice, possibly in a follow-up after this PR proved that it works as expected :)